scripts/operations/configuration/python_lib/api_requests.py

```diff
@@ -48,20 +61,80 @@ def log_error_raise_exception(msg: str, parent_exception: Exception = None) -> N
     raise common.ScriptException(msg) from parent_exception
 
 
-def make_api_request_with_retries(request_method: Callable,
-                                  url: str,
-                                  **request_kwargs) -> requests.models.Response:
+def get_admin_client_token(k8s_client: k8s.V1Client) -> str:
+    """
+    Return the token string needed to get API token
+    """
+    secret_label = f"{ADMIN_CLIENT_SECRET_NAMESPACE}/{ADMIN_CLIENT_SECRET_NAME} Kubernetes secret"
+    admin_client_secret_data = k8s_client.get_secret_data(name=ADMIN_CLIENT_SECRET_NAME,
+                                                          namespace=ADMIN_CLIENT_SECRET_NAMESPACE)
+    try:
+        encoded_client_secret = admin_client_secret_data["client-secret"]
+    except (KeyError, TypeError) as exc:
+        log_error_raise_exception(
+            f"{secret_label} has an unexpected format", exc)
+    # return decoded bytes as a string
+    try:
+        return base64.standard_b64decode(encoded_client_secret).decode()
```

Comment:
My IDE complains that `encoded_client_secret` may be referenced before assignment, but that's only because it can't tell that `log_error_raise_exception` always raises an exception. Nothing to necessarily fix here, just wanted to point it out.

Reply:
There are a lot of instances of this in the code. I'm not worrying about it for now.

Reply:
Yeah, I noticed the same complaint and for the same reason. Do you think it is worth assigning it a null value just to avoid the linting complaint?

Reply:
Nah, I don't think it's worth addressing unless we change how the error handling is done more fundamentally.